### PR TITLE
Improve functionality of Hazard.from_raster_xarray

### PR DESCRIPTION
**Pull request by peanutfun**
Changes proposed in this PR:
- Repeat single-valued data
- Set default fraction to `None`, which complies with the new `__init__` signature
- Update docs
- Update tests

The changes of this PR help me implement my GloFAS river flood model, see https://github.com/CLIMADA-project/climada_petals/issues/60

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Source branch up-to-date with target branch
- [x] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [x] [Tests][testing] updated
- [x] [Tests][testing] passing
- [ ] No new [linter issues][linter]
   ❌  3 new LOW issues

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

**Comment by peanutfun** [comment]
@emanuel-schmid I finished working on this, it's ready for review! 🙌

File: climada/hazard/base.py [hunk 1]
```diff
@@ -743,10 +789,10 @@ def strict_positive_int_accessor(array: xr.DataArray) -> np.ndarray:
                 # The accessor for the data in the Dataset
                 accessor=[
                     lambda x: to_csr_matrix(default_accessor(x)),
-                    default_accessor,
-                    strict_positive_int_accessor,
-                    lambda x: list(default_accessor(x).flat),  # list, not np.array
+                    lambda x: maybe_repeat(default_accessor(x), num_events),
                     strict_positive_int_accessor,
```
Comment (emanuel-schmid) [inline climada/hazard/base.py:793]:
```suggestion
                    lambda x: strict_positive_int_accessor(x),
```
for being aligned, mainly

File: climada/hazard/base.py [hunk 2]
```diff
@@ -850,7 +896,9 @@ def vshape(array):
                 return array.shape
 
             # Check size for read data
-            if not np.array_equal(vshape(val), vshape(default_value)):
+            if default_value is not None and (
+                not np.array_equal(vshape(val), vshape(default_value))
+            ):
```
Comment (emanuel-schmid) [inline climada/hazard/base.py:901]:
```suggestion
            if (default_value is not None
               and not np.array_equal(vshape(val), vshape(default_value))
            ):
```

Comment (peanutfun) [reply climada/hazard/base.py:901]: I thought I had this formatted with black but this was not the case. So I did it again and black wants to format it this way:
```suggestion
            if default_value is not None and not np.array_equal(
                vshape(val), vshape(default_value)
            ):
```
Can we simply agree on black? 😅

File: climada/hazard/base.py (hunk 1 above)
Comment (peanutfun) [reply climada/hazard/base.py:793]: It's more "symmetrical", but the lambda here is completely superfluous, so I would like to leave it out 😬 You decide!

**Pull request merged**